test/e2e/autoscale_test.go

```diff
@@ -143,7 +138,7 @@ func TestAutoscaleSustaining(t *testing.T) {
 				}))
 			test.EnsureTearDown(t, ctx.Clients(), ctx.Names())
 
-			AssertAutoscaleUpToNumPods(ctx, 1, 10, time.After(2*time.Minute), false /* quick */)
+			AssertAutoscaleUpToNumPods(ctx, 1, 8, time.After(2*time.Minute), false /* quick */)
```

Comment:
If flakiness is only for the exponential case we could only adjust target for that to distinguish between the two algorithms for the same traffic generated.

Reply:
Hmm, I'd probably prefer setting the same target for both because we're adjusting this for environment that is not able to achieve what is requested by the test (the test requests certain number of workers that run in parallel). 
For environments which can satisfy the test requirements, the different target is not necessary. So, it complicates things for the more general case as well. But not sure.

Reply:
> afaik exponential favors latest values in the window statistics

In that case, would frequent changes in traffic also cause more frequent scaling up/down with the exponential algorithm than with the linear one?

Reply:
I wonder if it would be better to set different targets depending on environment, using the short flag to distinguish the constrained one (i.e. something like `targetPods := 10; if testing.Short() { targetPods = 8 }`)? That said, I'm not sure if there was a specific reason why we chose 10 to begin with or if it was just a nice round number.

(The weird thing is that these tests _used_ to work on Kind too. If I remember correctly, it was right around the time that they switched over to systemd cgroups driver that things stopped working. But that's neither here nor there...)

Reply:
> I wonder if it would be better to set different targets depending on environment, using the short flag to distinguish the constrained one (i.e. something like targetPods := 10; if testing.Short() { targetPods = 8 })?

That's an option too. But having 10 or 8 doesn't make a reasonable difference to me. As you said, it's not clear where the value 10 came from. So, I chose one common value (8) to simplify the test and have it identical for all environments.

Reply:
I think it's probably fine to just go with 8 (I don't know what scaling up two more pods would really tell us, other than we can handle double digits), but will give @dprotaso a chance to weigh in in case he has more context here.

Can you also drop the `-short` flag here: https://github.com/knative/serving/blob/9b9a9518d8074e47e57bc62e604c8d27f9747a1b/.github/workflows/kind-e2e.yaml#L331 ? We added that just for this test, so may as well get rid of it if we're fixing the issue that caused us to add it

Reply:
@dprotaso gentle ping.